unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/HighlightsSubSection/HighlightsSubSectionMenu/HighlightsSubSectionComponentController.cs

```diff
@@ -130,63 +134,66 @@ internal void RequestAllPlacesAndEvents()
 
     public void RequestAllFromAPI()
     {
-        cts?.Cancel();
-        cts?.Dispose();
-        cts = new CancellationTokenSource();
-        placesAPIApiController.GetAllPlacesFromPlacesAPI(
-            OnCompleted: (placeList, total) =>
-            {
-                placesFromAPI = placeList;
-
-                eventsAPIApiController.GetAllEvents(
-                    OnSuccess: eventList =>
-                    {
-                        eventsFromAPI = eventList;
-                        OnRequestedPlacesAndEventsUpdated();
-                    },
-                    OnFail: error =>
-                    {
-                        OnRequestedPlacesAndEventsUpdated();
-                        Debug.LogError($"Error receiving events from the API: {error}");
-                    });
-            }, 0, 20, cts.Token);
+        requestAllCts?.SafeCancelAndDispose();
+        requestAllCts = CancellationTokenSource.CreateLinkedTokenSource(disposeCts.Token);
+        RequestAllFromApiAsync(requestAllCts.Token).Forget();
+
+    }
+
+    private async UniTask RequestAllFromApiAsync(CancellationToken ct)
```

Comment:
I think you should catch `OperationCanceled` at least.

Reply:
Done!